tools/src/url_map_handler.rs

```diff
@@ -0,0 +1,21 @@
+use crate::file_format::url_map::{ URLMap, URLMapItem, read_url_map };
+use crate::file_format::analysis_manglings::mangle_file;
+
+static mut URL_MAP_PATH: Option<String> = None;
+
+pub fn set_url_map_path(path: &str) {
+    unsafe {
+        URL_MAP_PATH = Some(path.to_string());
+    }
+}
+
+pub fn get_file_paths_for_url(url: &str) -> Option<Vec<URLMapItem>> {
+    lazy_static! {
+        static ref URL_MAP: URLMap = unsafe {
+            read_url_map(URL_MAP_PATH.as_ref().unwrap().as_str())
+        };
+    }
+
+    let sym = format!("URL_{}", mangle_file(url));
```

Comment:
should this be `FILE_`?

Reply:
This is the key in the `url-map.json` data, which is a map from `URL_*` symbol to file data (pretty string and `FILE_*` symbol).
I'll add comment.

Reply:
Oh, right, I totally missed the next line was doing the lookup transform.  A comment certainly wouldn't hurt but this was just a case of me going through things too fast.